libsplinter/src/network/peer_manager/peer_map.rs

```diff
@@ -0,0 +1,383 @@
+    //  2. Insert peer test_peer and veirfy id is in self.peers
+    //  3. Verify that the correct active endpoint is returned when removing test_peer
+    #[test]
+    fn test_remove_peer() {
+        let mut peer_map = PeerMap::new();
+
+        let active_endpoint = peer_map.remove("test_peer");
+
+        assert_eq!(active_endpoint, None,);
+
+        peer_map.insert(
+            "test_peer".to_string(),
+            vec!["test_endpoint1".to_string(), "test_endpoint2".to_string()],
+            "test_endpoint2".to_string(),
+        );
+        assert!(peer_map.peers.contains_key("test_peer"));
+
+        let active_endpoint = peer_map.remove("test_peer");
+        assert!(!peer_map.peers.contains_key("test_peer"));
+
+        assert_eq!(active_endpoint, Some("test_endpoint2".to_string()),);
+    }
+
+    // Test that update_peer_id() works correctly
+    //  1. Test that an error is returned if the old peer id does not exist
+    //  2. Insert test_peer and check it is in self.peers
+    //  3. Update test_peer to have the id new_peer
+    //  4. Verify that peers contains new_peer and redirects contains a redirect from test_peer
+    //     to new_peer
+    #[test]
+    fn test_update_peer_id() {
+        let mut peer_map = PeerMap::new();
+
+        if let Ok(()) = peer_map.update_peer_id("test_peer".to_string(), "new_peer".to_string()) {
+            panic!("Should not be able to update peer because old peer does not exist")
+        }
+
+        peer_map.insert(
+            "test_peer".to_string(),
+            vec!["test_endpoint1".to_string(), "test_endpoint2".to_string()],
+            "test_endpoint2".to_string(),
+        );
+        assert!(peer_map.peers.contains_key("test_peer"));
+
+        peer_map
+            .update_peer_id("test_peer".to_string(), "new_peer".to_string())
+            .expect("Unable to update peer id");
+
+        assert!(peer_map.peers.contains_key("new_peer"));
+        assert!(peer_map.redirects.contains_key("test_peer"));
+    }
+
+    // Test that a peer can be updated
+    //  1. Check that an error is returned if the peer does not exist
+    //  2. Insert test_peer with active endpoint test_endpoint2
+    //  3. Update the active enpdoint for test_peer to test_endpoint1 and set the status to
+    //     disconnected
+    //  4. Check that the peer's metadata now points to test_endpoint1 and the peer is disconnected
+    #[test]
+    fn test_get_update_active_endpoint() {
+        let mut peer_map = PeerMap::new();
+        let no_peer_metadata = PeerMetadata {
+            id: "test_peer".to_string(),
+            endpoints: vec!["test_endpoint1".to_string(), "test_endpoint2".to_string()],
+            active_endpoint: "test_endpoint1".to_string(),
+            status: PeerStatus::Connected,
+        };
+
+        if let Ok(()) = peer_map.update_peer(no_peer_metadata) {
+            panic!("Should not be able to update peer because test_peer does not exist")
```

Comment:
"Should not" seems odd in a panic message - this word implies that the reader should (or in this case, should not) take some action. Perhaps change to "Cannot update..."?

Reply:
Never mind. This wording is consistent with similar panic messages in other places.  I'll dismiss this comment.